backend/modules/model_gateway/model_gateway.py

```diff
@@ -73,44 +66,79 @@ def get_embedder_from_model_config(self, model_name: str) -> Embeddings:
         model_provider_config: ModelProviderConfig = self.model_name_to_provider_config[
             model_name
         ]
-        if not model_provider_config.api_key_env_var:
-            api_key = "EMPTY"
+
+        if model_provider_config.provider_name == "sambastudio":
+            return SambaStudioEmbeddings()
         else:
-            api_key = os.environ.get(model_provider_config.api_key_env_var, "")
-        model_id = "/".join(model_name.split("/")[1:])
-        return OpenAIEmbeddings(
-            openai_api_key=api_key,
-            model=model_id,
-            openai_api_base=model_provider_config.base_url,
-            check_embedding_ctx_length=(
-                model_provider_config.provider_name == "openai"
-            ),
-        )
+            api_key = (
+                os.environ.get(model_provider_config.api_key_env_var, "")
+                if model_provider_config.api_key_env_var
+                else "EMPTY"
+            )
+            model_id = "/".join(model_name.split("/")[1:])
+            return OpenAIEmbeddings(
+                openai_api_key=api_key,
+                model=model_id,
+                openai_api_base=model_provider_config.base_url,
+                check_embedding_ctx_length=(
+                    model_provider_config.provider_name == "openai"
+                ),
+            )
 
     def get_llm_from_model_config(
         self, model_config: ModelConfig, stream=False
-    ) -> BaseChatModel:
+    ) -> Union[BaseChatModel, LLM]:
         if model_config.name not in self.model_name_to_provider_config:
             raise ValueError(
                 f"Model {model_config.name} not registered in the model gateway."
             )
         model_provider_config: ModelProviderConfig = self.model_name_to_provider_config[
             model_config.name
         ]
+
         if not model_config.parameters:
             model_config.parameters = {}
-        if not model_provider_config.api_key_env_var:
-            api_key = None
-        else:
-            api_key = os.environ.get(model_provider_config.api_key_env_var, "")
-        model_id = "/".join(model_config.name.split("/")[1:])
-        return ChatOpenAI(
-            model=model_id,
-            temperature=model_config.parameters.get("temperature", 0.1),
-            streaming=stream,
-            api_key=api_key,
-            base_url=model_provider_config.base_url,
+
+        api_key = (
+            os.environ.get(model_provider_config.api_key_env_var, "")
+            if model_provider_config.api_key_env_var
+            else None
         )
+        model_id = "/".join(model_config.name.split("/")[1:])
+
+        if model_provider_config.provider_name in [
+            "openai",
+            "local-ollama",
+            "local-infinity",
+            "truefoundry",
+        ]:
```

Comment:
@snova-kwasia : Does Samba studio not provide openai compatible APIs? If thats the case, it will be the easiest - else we should introduce it in the api_format flag - not whitelist the providers.